x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/mapper/AggregateMetricFieldMapper.java

```diff
@@ -0,0 +1,340 @@
+        min, max, sum, value_count;
+    }
+
+    public static class Defaults {
+        public static final Explicit<Boolean> IGNORE_MALFORMED = new Explicit<>(false, false);
+        public static final Explicit<Set<String>> METRICS = new Explicit<>(Collections.emptySet(), false);
+        public static final AggregateMetricFieldType FIELD_TYPE = new AggregateMetricFieldType();
+
+        static {
+            FIELD_TYPE.setTokenized(false);
+            FIELD_TYPE.setHasDocValues(true);
+            FIELD_TYPE.setIndexOptions(IndexOptions.NONE);
+            FIELD_TYPE.freeze();
+        }
+    }
+
+    public static class Builder extends FieldMapper.Builder<AggregateMetricFieldMapper.Builder, AggregateMetricFieldMapper> {
+
+        private Boolean ignoreMalformed;
+
+        private Set<String> metrics;
+
+        public Builder(String name) {
+            super(name, Defaults.FIELD_TYPE, Defaults.FIELD_TYPE);
+            builder = this;
+        }
+
+        public AggregateMetricFieldMapper.Builder ignoreMalformed(boolean ignoreMalformed) {
+            this.ignoreMalformed = ignoreMalformed;
+            return builder;
+        }
+
+        protected Explicit<Boolean> ignoreMalformed(BuilderContext context) {
+            if (ignoreMalformed != null) {
+                return new Explicit<>(ignoreMalformed, true);
+            }
+            if (context.indexSettings() != null) {
+                return new Explicit<>(IGNORE_MALFORMED_SETTING.get(context.indexSettings()), false);
+            }
+            return AggregateMetricFieldMapper.Defaults.IGNORE_MALFORMED;
+        }
+
+        public AggregateMetricFieldMapper.Builder metrics(Set<String> metrics) {
+            this.metrics = metrics;
+            return builder;
+        }
+
+        protected Explicit<Set<String>> metrics(BuilderContext context) {
+            if (metrics != null) {
+                return new Explicit<>(metrics, true);
+            }
+            return Defaults.METRICS;
+        }
+
+        @Override
+        public AggregateMetricFieldMapper build(BuilderContext context) {
+            setupFieldType(context);
+            return new AggregateMetricFieldMapper(name, fieldType, defaultFieldType,
+                context.indexSettings(), multiFieldsBuilder.build(this, context),
+                ignoreMalformed(context), metrics(context), copyTo);
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
+
+        @Override
+        public Mapper.Builder<Builder, AggregateMetricFieldMapper> parse(String name,
+                                                                         Map<String, Object> node,
+                                                                         ParserContext parserContext) throws MapperParsingException {
+            AggregateMetricFieldMapper.Builder builder = new AggregateMetricFieldMapper.Builder(name);
+            TypeParsers.parseField(builder, name, node, parserContext);
```

Comment:
Let's not call parseField, none of the properties that it parses are supported by this field?

Reply:
Fixed